public/app/core/logs_model.ts

```diff
@@ -303,11 +303,37 @@ interface LogFields {
 
   timeField: FieldWithIndex;
   stringField: FieldWithIndex;
+  labelsField?: FieldWithIndex;
   timeNanosecondField?: FieldWithIndex;
   logLevelField?: FieldWithIndex;
   idField?: FieldWithIndex;
 }
 
+function getAllLabels(fields: LogFields): Labels[] {
+  // there are two types of dataframes we handle.
+  // either labels-on-the-string-field, or labels-in-the-labels-field
+
+  const { stringField, labelsField } = fields;
+
+  const fieldLabels = stringField.labels !== undefined ? [stringField.labels] : [];
+
+  const labelsFieldLabels: Labels[] = labelsField !== undefined ? labelsField.values.toArray() : [];
+
+  return [...fieldLabels, ...labelsFieldLabels];
```

Comment:
it can happen, like the current loki-live-stream code stores the stream-selector-labels (that will never change) in the string-field's `.labels` attribute, and the "other" labels in the separate labels-field